Generated unit tests fail with repeated double in response

https://github.com/googleapis/gapic-generator-python/pull/1541 fixed an issue with a float comparison in a unit test however the fix does not work for repeated fields.

An example of the error is 
```
>       assert math.isclose(response.repeatedfield, [0.1809], rel_tol=1e-6)
E       TypeError: must be real number, not Repeated
```

I can recreate the issue by adding a fragment test with a `repeated double` field in the response.
```
=================================== FAILURES ===================================
________________________ test_my_method[MethodRequest] _________________________

request_type = <class 'google.fragment.types.test_response_has_repeated_double.MethodRequest'>
transport = 'grpc'

    @pytest.mark.parametrize("request_type", [
      test_response_has_repeated_double.MethodRequest,
      dict,
    ])
    def test_my_method(request_type, transport: str = 'grpc'):
        client = MyServiceClient(
            credentials=ga_credentials.AnonymousCredentials(),
            transport=transport,
        )
    
        # Everything is optional in proto3 as far as the runtime is concerned,
        # and we are mocking out the actual API, so just send an empty request.
        request = request_type()
    
        # Mock the actual call within the gRPC stub, and fake the request.
        with mock.patch.object(
                type(client.transport.my_method),
                '__call__') as call:
            # Designate an appropriate return value for the call.
            call.return_value = test_response_has_repeated_double.MethodResponse(
                result=[0.671],
            )
            response = client.my_method(request)
    
            # Establish that the underlying gRPC stub method was called.
            assert len(call.mock_calls) == 1
            _, args, _ = call.mock_calls[0]
            assert args[0] == test_response_has_repeated_double.MethodRequest()
    
        # Establish that the response is the type that we expect.
        assert isinstance(response, test_response_has_repeated_double.MethodResponse)
>       assert math.isclose(response.result, [0.671], rel_tol=1e-6)
E       TypeError: must be real number, not Repeated

/tmp/tmpq09qwuah/tests/unit/gapic/fragment/test_my_service.py:572: TypeError
```

--- a/gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
+++ b/gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
@@ -587,7 +587,16 @@ def test_{{ method_name }}(request_type, transport: str = 'grpc'):
     {% for field in method.output.fields.values() | rejectattr('message') %}
     {% if not field.oneof or field.proto3_optional %}
     {% if field.field_pb.type in [1, 2] %}{# Use approx eq for floats #}
+    {% if field.repeated %}
+    for index in range(len(response.{{ field.name }})):
+        assert math.isclose(
+            response.{{ field.name }}[index],
+            {{ field.mock_value }}[index],
+            rel_tol=1e-6,
+        )
+    {% else %}{# field.repeated #}
     assert math.isclose(response.{{ field.name }}, {{ field.mock_value }}, rel_tol=1e-6)
+    {% endif %}{# field.repeated #}
     {% elif field.field_pb.type == 8 %}{# Use 'is' for bools #}
     assert response.{{ field.name }} is {{ field.mock_value }}
     {% else %}
@@ -1076,7 +1085,16 @@ def test_{{ method.name|snake_case }}_rest(request_type):
     {% for field in method.output.fields.values() | rejectattr('message') %}
     {% if not field.oneof or field.proto3_optional %}
     {% if field.field_pb.type in [1, 2] %}{# Use approx eq for floats #}
+    {% if field.repeated %}
+    for index in range(len(response.{{ field.name }})):
+        assert math.isclose(
+            response.{{ field.name }}[index],
+            {{ field.mock_value }}[index],
+            rel_tol=1e-6,
+        )
+    {% else %}{# field.repeated #}
     assert math.isclose(response.{{ field.name }}, {{ field.mock_value }}, rel_tol=1e-6)
+    {% endif %}
     {% elif field.field_pb.type == 8 %}{# Use 'is' for bools #}
     assert response.{{ field.name }} is {{ field.mock_value }}
     {% else %}

--- a/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2
+++ b/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2
@@ -76,7 +76,16 @@ def test_{{ method_name }}(request_type, transport: str = 'grpc'):
     {% for field in method_output.fields.values() | rejectattr('message') %}
     {% if not field.oneof or field.proto3_optional %}
     {% if field.field_pb.type in [1, 2] %}{# Use approx eq for floats #}
+    {% if field.repeated %}
+    for index in range(len(response.{{ field.name }})):
+        assert math.isclose(
+            response.{{ field.name }}[index],
+            {{ field.mock_value }}[index],
+            rel_tol=1e-6,
+        )
+    {% else %}{# field.repeated #}
     assert math.isclose(response.{{ field.name }}, {{ field.mock_value }}, rel_tol=1e-6)
+    {% endif %}{# field.repeated #}
     {% elif field.field_pb.type == 8 %}{# Use 'is' for bools #}
     assert response.{{ field.name }} is {{ field.mock_value }}
     {% else %}
@@ -187,7 +196,16 @@ async def test_{{ method_name }}_async(transport: str = 'grpc_asyncio', request_
     {% for field in method_output.fields.values() | rejectattr('message') %}
     {% if not field.oneof or field.proto3_optional %}
     {% if field.field_pb.type in [1, 2] %}{# Use approx eq for floats #}
+    {% if field.repeated %}
+    for index in range(len(response.{{ field.name }})):
+        assert math.isclose(
+            response.{{ field.name }}[index],
+            {{ field.mock_value }}[index],
+            rel_tol=1e-6,
+        )
+    {% else %}{# field.repeated #}
     assert math.isclose(response.{{ field.name }}, {{ field.mock_value }}, rel_tol=1e-6)
+    {% endif %}{# field.repeated #}
     {% elif field.field_pb.type == 8 %}{# Use 'is' for bools #}
     assert response.{{ field.name }} is {{ field.mock_value }}
     {% else %}
@@ -958,7 +976,16 @@ def test_{{ method_name }}_rest(request_type):
     {% for field in method_output.fields.values() | rejectattr('message') %}
     {% if not field.oneof or field.proto3_optional %}
     {% if field.field_pb.type in [1, 2] %}{# Use approx eq for floats #}
+    {% if field.repeated %}
+    for index in range(len(response.{{ field.name }})):
+        assert math.isclose(
+            response.{{ field.name }}[index],
+            {{ field.mock_value }}[index],
+            rel_tol=1e-6,
+        )
+    {% else %}{# field.repeated #}
     assert math.isclose(response.{{ field.name }}, {{ field.mock_value }}, rel_tol=1e-6)
+    {% endif %}{# field.repeated #}
     {% elif field.field_pb.type == 8 %}{# Use 'is' for bools #}
     assert response.{{ field.name }} is {{ field.mock_value }}
     {% else %}

--- a/tests/fragments/test_repeated_double.proto
+++ b/tests/fragments/test_repeated_double.proto
@@ -1,0 +1,38 @@
+// Copyright (C) 2021  Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package google.fragment;
+
+import "google/protobuf/struct.proto";
+import "google/api/client.proto";
+
+service MyServiceRepeatedDouble {
+  option (google.api.default_host) = "my.example.com";
+
+  rpc MyMethod(MethodRequestWithRepeatedDouble) returns (MethodResponseWithRepeatedDouble) {
+    option (google.api.method_signature) = "parameter,items";
+  }
+}
+
+message MethodRequestWithRepeatedDouble {
+  google.protobuf.Value parameter = 1;
+  repeated google.protobuf.Value items = 2;
+  repeated double repeated_items = 3;
+}
+
+message MethodResponseWithRepeatedDouble {
+  repeated double result = 1;
+}

